boards/recovery/src/data_manager.rs

```diff
@@ -0,0 +1,105 @@
+use heapless::spsc::Queue;
+use messages::Message;
+use messages::sensor::{Air, EkfNav1, EkfNav2, EkfQuat, GpsVel, Imu1, Imu2, UtcTime};
+use crate::state_machine::RocketStates;
+
+const MAIN_HEIGHT: f32 = 450.0;
+const ACCEL_MIN: f32 = 20.0;
+
+pub struct DataManager {
+    pub sbg_air: Option<Air>,
+    pub sbg_nav1: Option<EkfNav1>,
+    pub sbg_nav2: Option<EkfNav2>,
+    pub sbg_quat: Option<EkfQuat>,
+    pub sbg_gps_vel: Option<GpsVel>,
+    pub sbg_imu1: Option<Imu1>,
+    pub sbg_imu2: Option<Imu2>,
+    pub sbg_utc_time: Option<UtcTime>,
+    pub historical_pressure: Queue<f32, 8>, // probably want to make this a ring buffer
+    pub current_state: Option<RocketStates>,
+}
+
+impl DataManager {
+    pub fn new() -> Self {
+        let historical_pressure = Queue::new();
+        Self {
+            sbg_air: None,
+            sbg_nav1: None,
+            sbg_nav2: None,
+            sbg_quat: None,
+            sbg_gps_vel: None,
+            sbg_imu1: None,
+            sbg_imu2: None,
+            sbg_utc_time: None,
+            historical_pressure,
+            current_state: None,
+        }
+    }
+    // Questions, questions, questions!!??? does it work!!!!!
+    // Probably not, but it's a start
+    pub fn is_falling(&self) -> bool {
```

Comment:
The idea is to start analyzing pressure when velocity is less than 10 m/s. This should allow us to avoid transonic flight and the complexity of handling pressure as it behaves when the vehicle is transonic.